src/Pos/Block/Error.hs

```diff
@@ -3,32 +3,82 @@
 module Pos.Block.Error
        ( BlkError
        , RollbackException(..)
+       , ApplyBlocksException(..)
+       , VerifyBlocksException(..)
        ) where
 
 import           Universum
 
-import           Control.Exception    (Exception (..))
-import qualified Data.Text.Buildable
+import           Control.Exception      (Exception (..))
+import           Data.Text.Buildable    (Buildable (..))
+import           Data.Text.Lazy.Builder (Builder, fromText)
+import           Formatting             (bprint, stext, (%))
 
-import           Pos.Block.Logic.Util (tipMismatchMsg)
-import           Pos.Core             (HeaderHash)
+import           Pos.Core               (HeaderHash)
+import           Pos.Crypto             (shortHashF)
 
 data BlkError
 
 --instance Exception BlkError
 
 --instance Buildable BlkError
 
+-- | This function can be used to create a message when tip mismatch
+-- is detected (usually between tip stored in DB and some other tip
+-- received from somewhere).
+tipMismatchMsg :: Text -> HeaderHash -> HeaderHash -> Builder
+tipMismatchMsg action storedTip attemptedTip =
+    bprint
+        ("Can't "%stext%" block because of tip mismatch (stored is "
+         %shortHashF%", attempted is "%shortHashF%")")
+        action storedTip attemptedTip
+
 data RollbackException = RollbackTipMismatch HeaderHash HeaderHash
     deriving (Show)
 
-renderRollbackException :: RollbackException -> Text
+renderRollbackException :: RollbackException -> Builder
```

Comment:
I don't know what's the purpose of moving it into a separate function, but I don't mind.